### PR TITLE
fix bug: index out of bound in record StatsBuffer values

### PR DESCRIPTION
**Pull request by andyzhuangyy**
while `count` in StatsBuffer overflow, the array index will be out of bound.

```
public void record(long n) {
    values[count++ % size] = n;
    total += n;
    sumSquares += n * n;
 }
```

### CODE REVIEW

File: servo-core/src/main/java/com/netflix/servo/stats/StatsBuffer.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public void reset() {
    * Record a new value for this buffer.
    */
   public void record(long n) {
-    values[count++ % size] = n;
+    values[(int) Long.remainderUnsigned(count++, size)] = n;
```
Comment (brharrington) [inline servo-core/src/main/java/com/netflix/servo/stats/StatsBuffer.java]: Integer.remainderUnsigned seems more appropriate here and would not need a cast.

**Comment by brharrington** [comment]
Thanks for the patch. How hard do you think it would be to add a test case to cover the overflow scenario?

**Comment by andyzhuangyy** [comment]
I check the code again and find that if I want to fix the overflow defect, I should not only update the `record` method, but also `computeStats` method.

I should fix it at a higher perspective. And make all the logic reasonable.

So I need some advice about the strategy while overflow has happened (maybe caused by not calling `reset` method for a long time).

Looking forward to your help.

**Comment by brharrington** [comment]
I pulled in your changes and added some more in #385.
